aiida_vasp/assistant/parameters.py

```diff
@@ -235,17 +234,19 @@ def _set_extra_vasp_parameters(self):
         to make sure it was valid input to the VASP workchain.
 
         """
-        try:
-            if self._parameters.dynamics:
-                self._massage.dynamics = AttributeDict()
-            for key, item in self._parameters.dynamics.items():
-                key = key.lower()
-                if key in ['positions_dof']:
-                    self._massage.dynamics[key] = item
-                else:
-                    break
-        except AttributeError:
-            pass
+        # return if dynamics is not supplied
+        if 'dynamics' not in self._parameters:
+            return
+
+        if self._parameters.dynamics:
+            self._massage.dynamics = AttributeDict()
+
+        for key, item in self._parameters.dynamics.items():
+            key = key.lower()
+            if key in ['positions_dof']:
+                self._massage.dynamics[key] = item
+            else:
+                break
```

Comment:
@espenfl This is `break` really necessary? If the key is invalid should it just skip or raise? If we have `break` the order of the keys will also affect the outcome...

Reply:
Yes, I think this was just done now since there are only one matching key so it does not really matter. Also, number of keys are minute now. Please generalize if you want. Thanks.